corehq/apps/reports/tasks.py

```diff
@@ -261,20 +261,20 @@ def send_email_report(self, recipient_emails, domain, report_slug, report_type,
 
     subject = cleaned_data['subject'] or _("Email report from CommCare HQ")
 
-    content = _render_report_configs(
-        mock_request, [config], domain, user_id, couch_user, True, lang=couch_user.language,
-        notes=cleaned_data['notes'], once=once
-    )[0]
-    body = render_full_report_notification(None, content).content
-
     try:
+        content = _render_report_configs(
+            mock_request, [config], domain, user_id, couch_user, True, lang=couch_user.language,
+            notes=cleaned_data['notes'], once=once
+        )[0]
+        body = render_full_report_notification(None, content).content
+
         for recipient in recipient_emails:
             send_HTML_email(subject, recipient,
                             body, email_from=settings.DEFAULT_FROM_EMAIL,
                             smtp_exception_skip_list=[LARGE_FILE_SIZE_ERROR_CODE])
 
     except Exception as er:
-        if getattr(er, 'smtp_code', None) == LARGE_FILE_SIZE_ERROR_CODE:
+        if getattr(er, 'smtp_code', None) == LARGE_FILE_SIZE_ERROR_CODE or type(er) == ESError:
```

Comment:
This `if` statement says it regenerates the report to email a download link if the report is too big to email, but this changes it to also do that if there's an ESError.  If it's an ES error, why would the report succeed on the second attempt but not the first?  Can we avoid generating all large reports twice in the first place?

This error handling is also pretty messy - is there a better error we can catch than `Exception` for the smtp issue?

Likely out of scope for this PR, but right now it looks like for any other exception, this will retry 10 times and then give up.  Does it ever notify us of the error?  Looks like this was copy/pasted from the email tasks here:
https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/tasks.py
But in this case it also generates a report, which is a very resource-intensive task, and this version doesn't `notify_exception`

Reply:
> If it's an ES error, why would the report succeed on the second attempt but not the first?
On the first attempt, it tried to send the report in the html body of the email, but this time around, it sends it as an excel attachment. This seemed like the simpler fix for the issue on ICDS (to send the report as an attachment if it fails, rather than defaulting to excel attachment if its over a certain size or something like that.)

Agreed that the handling is messy, I'm not really sure how to make the smtp logic cleaner without excepting the whole underlying exception (as opposed to just this error code), which I'm assuming was initially written like this for a reason... I'll gladly take any suggestions though

Reply:
> On the first attempt, it tried to send the report in the html body of the email, but this time around, it sends it as an excel attachment.

That's what I mean - if we get an `ESError`, then presumably we haven't even gotten to the part where it tries to send the report in the body of the email or as an attachment.  The failure is that we can't fetch the data for the report, so changing the way it's rendered seems like a non sequitor.

Reply:
Regarding the `except Exception` business - I'm okay with the first part, since it looks for a specific error code, meaning that we are catching the exception we expect, but the second part takes all other exceptions and retries them.  I think the second part should at least `notify_exception`.  My issue is that this change moves the report generation into this `try` block, so there's a very broad scope of failures that could be silenced.

What does this look like on ICDS right now?  Do we get any sort of error, or does emailing this report just appear to do nothing?